src/dash/DashAdapter.js

```diff
@@ -1092,8 +1109,27 @@ function DashAdapter() {
         }
 
         mediaInfo.isText = dashManifestModel.getIsText(realAdaptation);
-        mediaInfo.supplementalProperties = dashManifestModel.getSupplementalProperties(realAdaptation);
-
+        mediaInfo.supplementalProperties = dashManifestModel.getSupplementalPropertiesForAdaptation(realAdaptation);
```

Comment:
I am considering removing this "old" way of saving the data. With this we can only persist one element for each `schemeIdUri` as we use the `schemeIdUri` as a key. After a quick scan we only use `supplementalProperties` in the `PlaybackController` and that could be adjusted easily:

````
    function _checkEnableLowLatency(mediaInfo) {
        if (mediaInfo && mediaInfo.supplementalProperties &&
            mediaInfo.supplementalProperties[Constants.SUPPLEMENTAL_PROPERTY_DVB_LL_SCHEME] === 'true') {
            logger.debug('Low Latency critical SupplementalProperty set: Enabling low Latency');
            lowLatencyModeEnabled = true;
        }
    }
````

My suggestion would be to save the data as it is done below for `supplementalPropertiesAsArray`but in `supplementalProperties` and not use `supplementalPropertiesAsArray`

We can highlight this change in the release notes in case applications use this attribute in some way. However, the question arises if then do it all at once and remove the old format for `viewpoint`, `role`, `accessibility`, `audioChannelConfiguration` already here as well.

Reply:
Let's discuss tomorrow!

Just as a note:
If we conclude to abandon the "old way" for all DescriptorType elements, there are more to be changed, e.g. all the `matchSettings*()` in `MediaController.js`

Reply:
During our monthly call we agreed to include the breaking changes in v.5.0.0